### PR TITLE
Client Server implementation - sockets

### PR DESCRIPTION
**Pull request by Boogami**

### CODE REVIEW

File: server/server.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#include "server.h"
+
+void socketThread(int clientSocket)
+{
+    char client_message[2000];
```
Comment (Muratura43) [inline server/server.c]: Client message is 2000, server buffer is 1024 and client buffer is 512.
This is confusing. They should all be the same size.

File: lib/parser/parser.h [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+#include <unistd.h>
+#include <ftw.h>
+
+#include <errno.h>
+
+#include <sys/stat.h>
+#include <sys/socket.h>
+
+#include <netinet/in.h>
+#include <arpa/inet.h>
```
Comment (Muratura43) [inline lib/parser/parser.h]: Why do you need network libraries in the parser?
All network related should be contained in either the client or server.

**Comment by Muratura43** [comment]
Remove binary files from the build folder.

File: server/server.c [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    pthread_mutex_unlock(&lock);
+    sleep(1);
+    send(newSocket, buffer, 13, 0);
+    printf("Exit socketThread \n");
+    close(newSocket);
+}
+
+int main()
+{
+    short serverPort = 8080;
```
Comment (Muratura43) [inline server/server.c]: The port and IP should be common for the client and server.

File: client/client.c [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            //reset beginning time
+            gettimeofday(&begin, NULL);
+        }
+    }
+    return total_size;
+}
+
+void *clientThread(void *arg)
+{
+    short port = 8080;
```
Comment (Muratura43) [inline client/client.c]: Use a less generic port number to not have possible conflicts with the system.

**Review by Muratura43 — CHANGES_REQUESTED** [review]

File: server/server.c (hunk 1 above)
Comment (Boogami) [reply server/server.c]: changed to #define BUFF_SIZE 2048

File: lib/parser/parser.h (hunk 2 above)
Comment (Boogami) [reply lib/parser/parser.h]: removed network libraries in the parser

File: server/server.c (hunk 3 above)
Comment (Boogami) [reply server/server.c]: Done

File: client/client.c (hunk 4 above)
Comment (Boogami) [reply client/client.c]: Used 49152 port address.

**Review by Muratura43 — APPROVED** [review]

**Review by Muratura43 — APPROVED** [review]

**Pull request merged**